asv_bench/benchmarks/utils.py

```diff
@@ -572,3 +573,46 @@ def prepare_io_data(test_filename: str, data_type: str, shapes: list):
         df.to_csv(test_filenames[shape_id], index=False)
 
     return test_filenames
+
+
+CONFIG_FROM_FILE = None
+
+
+def get_benchmarks_shapes(bench_id: str, default: list):
+    """
+    Get custom benchmarks shapes from a json file stored in MODIN_ASV_DATASIZE_CONFIG.
+
+    If `bench_id` benchmark is not found in the file, then the default value will
+    be used.
+
+    Parameters
+    ----------
+    bench_id : str
+        Unique benchmark identifier that is used to get shapes.
+    default : list
+        Default shapes.
+
+    Returns
+    -------
+    list
+        Benchmark shapes.
+    """
+    try:
```

Comment:
Why do we need this `try` block?

Reply:
This is necessary to measure older commits than the one after the merge.

Reply:
Not sure how we can measure older commits if this file doesn't exist in the ones. Can you elaborate?

Reply:
Our benchmarking works as follows - the latest version of the Modin is taken. From it we only need `asv_bench` folder. When starting the benchmarks, an independent environment is created, in which the Modin of the version that we specified is installed.

That is, we use ASV configurations and tests from one place, and the executable code from another.

Reply:
Oh, I see, thanks!

Reply:
This should (really, even a separate PR just for that is fine) be added as a comment in this particular place. I remember asking the same question before! 😄